packages/repo-tools/src/commands/openapi/lint.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Spectral,
+  Document,
+  Ruleset,
+  RulesetDefinition,
+} from '@stoplight/spectral-core';
+import { Yaml } from '@stoplight/spectral-parsers';
+import ruleset from '@apisyouwonthate/style-guide';
+import fs from 'fs-extra';
+import chalk from 'chalk';
+import { resolve } from 'path';
+import { runner } from './runner';
+import { YAML_SCHEMA_PATH } from './constants';
+import { oas } from '@stoplight/spectral-rulesets';
+import { DiagnosticSeverity } from '@stoplight/types';
+import { pretty } from '@stoplight/spectral-formatters';
+
+async function lint(
+  directoryPath: string,
+  config?: { skipMissingYamlFile: boolean; strict: boolean },
+) {
+  const { skipMissingYamlFile, strict } = config ?? {};
+  const openapiPath = resolve(directoryPath, YAML_SCHEMA_PATH);
+  if (!(await fs.pathExists(openapiPath))) {
+    if (skipMissingYamlFile) {
+      return;
+    }
+    throw new Error(`Could not find ${YAML_SCHEMA_PATH} in root of directory.`);
+  }
+  const openapiFileContent = await fs.readFile(openapiPath, 'utf8');
+
+  const document = new Document(openapiFileContent, Yaml, openapiPath);
+
+  const spectral = new Spectral();
+
+  const backstageRuleset = new Ruleset(
+    {
+      extends: [oas, ruleset],
+      overrides: [
+        {
+          files: ['*'],
+          rules: {
+            'api-health': 'off',
+            'api-home': 'off',
+            'api-home-get': 'off',
+            'operation-tags': 'off',
+            'hosts-https-only-oas3': 'off',
+            'no-unknown-error-format': 'off',
```

Comment:
Makes me wonder if some form of `.spectralrc` or so is in our future ... since the fixed set of rules might not always be enough

Reply:
They support a YAML file, https://docs.stoplight.io/docs/spectral/01baf06bdd05a-rulesets. I wanted to stick with the Javascript-based interface, but would be straightforward to use their config file instead. Thoughts?

Reply:
Sounds good. Can be left as a future addition while we have sensible defaults in the mean time. But since this is going to run out there in the wild and not just in the repo we control, I bet someone is going to ask for it

Reply:
That's fair, it should be straightforward-ish to [pull in the file](https://docs.stoplight.io/docs/spectral/eb68e7afd463e-spectral-in-java-script#load-a-jsonyaml-ruleset) as well, probably a small effort to handle the merging of configs.